ruby/ql/lib/codeql/ruby/ast/Erb.qll

```diff
@@ -159,7 +159,7 @@ class ErbDirective extends TDirectiveNode, ErbAstNode {
    */
   Stmt getAChildStmt() {
     this.containsStmtStart(result) and
-    not this.containsStmtStart(result.getParent())
+    not this.containsStmtStart(parent*(result).getParent())
```

Comment:
Why this unrelated change?

Reply:
It's not unrelated. One of the synthesized variable accesses has a SimpleParameter as parent and because SimpleParameter is not longer a Stmt( Expr extends Stmt) this negation suddenly held. The `parent*` change is to find the nearest parent that is instanceof `Stmt`.

Reply:
What if we generalize `containsStmtStart(Stmt s)` to `containsNodeStart(AstNode n)` instead?

Reply:
That should work too.